src/main/java/edu/illinois/cs/dt/tools/utility/iFixPlusPlugin.java

```diff
@@ -0,0 +1,744 @@
+                            setProperty("statecapture.state", "failing_order");
+                    System.out.println("~~~~~~ Begin capturing the state in failing order!!");
+                    try {
+                        runner.runList(testFailOrder());
+                    }
+                    catch (Exception e) {
+                        System.out.println("## Encounter error in phase capturing the state in failing order!! " + e);
+
+                    }
+                    timing(startTime);
+                    startTime = System.currentTimeMillis();
+                    System.out.println("~~~~~~ Finish phase capturing the state in failing order!!");
+                } else {
+                    Files.write(Paths.get(output),
+                            "0,".getBytes(),
+                            StandardOpenOption.APPEND);
+                }
+
+                // phase 5: do the diff
+                System.out.println("~~~~~~ Begin diffing between the passing order and failing order!!!");
+
+                xmlFileNum = countDirNums(subxmlFold);
+                System.out.println("xmlFileNum: " + xmlFileNum);
+
+                File passingSubXmlFolder = new File(subxmlFold + "/failing_order_xml");
+                if (passingSubXmlFolder.exists() && failingSubXmlFolder.exists()) {
+                    try {
+                        diffing();
+                    }
+                    catch (Exception e) {
+                        System.out.println("## Encounter error in doing diffing: " + e);
+                    }
+                }
+                else {
+                    System.out.println("## Cannot do diff, xml files are not complete!");
+                }
+
+                timing(startTime);
+                startTime = System.currentTimeMillis();
+                System.out.println("~~~~~~ Finish diffing between the passing order and failing order!!!");
+
+                // output of phase 5
+                String diffFile = diffFieldsFold + "/diffFields.txt";
+
+                //create the reflection file
+                File reflectionFile = new File(reflectionFold + "/reflection.txt");
+                reflectionFile.createNewFile();
+
+                System.out.println("~~~~~~ Begin loading!");
+
+                // reflect at the after state
+                boolean reflectAfterOneSuccess = reflectEachField(diffFile, reflectionFile, runner, lastPolluter());
+                if (reflectAfterOneSuccess) {
+                    String successfulField = "";
+                    try (BufferedReader br = new BufferedReader(new FileReader(reflectionFile))) {
+                        String line;
+                        while ((line = br.readLine()) != null) {
+                            if (line.contains(" made test success#######")) {
+                                successfulField = line.substring(8, line.lastIndexOf(" made test success#######"));
```

Comment:
This feels so complicated to parse some English, can we get it to simply write out the valid polluted field?